flink-end-to-end-tests/flink-end-to-end-tests-common-s3/src/main/java/org/apache/flink/tests/util/s3/S3Resource.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.tests.util.s3;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.tests.util.util.FactoryUtils;
+import org.apache.flink.util.ExternalResource;
+
+import com.amazonaws.services.s3.AmazonS3;
+
+import java.net.URI;
+import java.util.Optional;
+
+/**
+ * Generic interface for interacting with S3.
+ */
+public interface S3Resource extends ExternalResource {
+	/**
+	 * Returns an S3 client that can be used to communicate with this resource.
+	 */
+	AmazonS3 getClient();
+
+	/**
+	 * The configurations option to add to the Flink application, such that this resource can be used through Flink.
```

Comment:
```suggestion
	 * The configuration options to add to the Flink application, such that this resource can be used through Flink.
```
This reads like these options are only added for a job submission ("application"), and not the cluster itself.